File: src/wfrp-helper-web/src/model/realm/RealmData.js
import Person from "../creature/Person";
import SpellSchool from "../knowledge/SpellSchool";
import ModelElement from "../ModelElement";
import Currency from "../world/Currency";
import Language from "../world/Language";
import Nation from "../world/Nation";
import Race from "../world/Race";
import Religion from "../world/Religion";

export default class RealmData extends ModelElement {
    nations: Nation[] = [];
    languages: Language[] = [];
    religions: Religion[] = [];
    currencies: Currency[] = [];
    races: Race[] = [];
    spellSchools: SpellSchool[] = [];
    persons: Person[] = [];


    updateWith(entity: RealmData): RealmData {
        this.updateEntityList(entity, 'nations', () => new Nation());
        this.updateEntityList(entity, 'languages', () => new Language());
        this.updateEntityList(entity, 'religions', () => new Religion());
        this.updateEntityList(entity, 'currencies', () => new Currency());
        this.updateEntityList(entity, 'races', () => new Race());
        this.updateEntityList(entity, 'spellSchools', () => new SpellSchool());
        this.updateEntityList(entity, 'persons', () => new Person());
        return this;
    }
}

File: src/main/java/pl/khuzzuk/wfrp/helper/service/realm/RealmData.java
package pl.khuzzuk.wfrp.helper.service.realm;

import lombok.Getter;
import lombok.Setter;
import pl.khuzzuk.wfrp.helper.model.creature.PersonDTO;
import pl.khuzzuk.wfrp.helper.model.knowledge.magic.SpellSchoolDTO;
import pl.khuzzuk.wfrp.helper.model.money.CurrencyDTO;
import pl.khuzzuk.wfrp.helper.model.world.NationDTO;
import pl.khuzzuk.wfrp.helper.model.world.RaceDTO;
import pl.khuzzuk.wfrp.helper.model.world.ReligionDTO;
import pl.khuzzuk.wfrp.helper.model.world.WorldLanguageDTO;

import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
public class RealmData {

  private List<NationDTO> nations = new ArrayList<>();
  private List<WorldLanguageDTO> languages = new ArrayList<>();
  private List<SpellSchoolDTO> spellSchools = new ArrayList<>();
  private List<ReligionDTO> religions = new ArrayList<>();
  private List<CurrencyDTO> currencies = new ArrayList<>();
  private List<RaceDTO> races = new ArrayList<>();
  private List<PersonDTO> persons = new ArrayList<>();
}
